### PR TITLE
Hana DB connectivity troubleshooting guide

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Create a troubleshooting guide for issues with connection to Hana DB

### CODE REVIEW

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    ```
+    Example output:
+    ```
+    HDBSQL version 2.20.20.1712178305, the SAP HANA Database interactive terminal.
+    Copyright 2000-2024 by SAP SE.
+    ```
+    3. Publish the image to a container registry.
+    4. Run the image in the Kubernetes cluster:
+    ```bash
+    kubectl create deployment hdbsql --image=<PUBLISHED_IMAGE_NAME>
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:52]:
```suggestion
    kubectl create deployment hdbsql --image={PUBLISHED_IMAGE_NAME}
```

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    ```
+    2. Test your image by running the following command:
+    ```bash
+    docker run --entrypoint "hdbsql" hdbsql -v
+    ```
+    Example output:
+    ```
+    HDBSQL version 2.20.20.1712178305, the SAP HANA Database interactive terminal.
+    Copyright 2000-2024 by SAP SE.
+    ```
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:48]:
```suggestion
2. To test your image, run the following command:
    ```bash
    docker run --entrypoint "hdbsql" hdbsql -v
    ```
    You get an output similar to this example:
    ```
    HDBSQL version 2.20.20.1712178305, the SAP HANA Database interactive terminal.
    Copyright 2000-2024 by SAP SE.
    ```
```

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    3. Publish the image to a container registry.
+    4. Run the image in the Kubernetes cluster:
+    ```bash
+    kubectl create deployment hdbsql --image=<PUBLISHED_IMAGE_NAME>
+    ```
+    5. Attach to the pod and try to connect to the Hana DB instance using the following command:
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:54]:
```suggestion
5. Attach to the Pod and try to connect to the SAP HANA Database instance using the following command:
```

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ```bash
+    hdbsql -n <HANA_DB_INSTANCE_ADDRESS> -u <HANA_DB_USER> -p <HANA_DB_PASSWORD>
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:56]:
```suggestion
    hdbsql -n {HANA_DB_INSTANCE_ADDRESS} -u {HANA_DB_USER} -p {HANA_DB_PASSWORD}
```

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    ```
+    6. If the connection is successful and you can execute queries, the issue is not related to setup of the cluster.
+    7. You might also want to check the connection from a pod that has Istio sidecar injected. In that case, create the deployment in a namespace with Istio sidecar injection enabled. The connection should still be successful.
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:59]:
```suggestion
7. Check the connection from a Pod that has the Istio sidecar injected. In that case, create the Deployment in a namespace with Istio sidecar injection enabled. The connection should be successful.
```

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    3. Publish the image to a container registry.
+    4. Run the image in the Kubernetes cluster:
+    ```bash
+    kubectl create deployment hdbsql --image=<PUBLISHED_IMAGE_NAME>
+    ```
+    5. Attach to the pod and try to connect to the Hana DB instance using the following command:
+    ```bash
+    hdbsql -n <HANA_DB_INSTANCE_ADDRESS> -u <HANA_DB_USER> -p <HANA_DB_PASSWORD>
+    ```
+    6. If the connection is successful and you can execute queries, the issue is not related to setup of the cluster.
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:58]:
```suggestion
6. If the connection is successful and you can execute queries, the issue is not related to the setup of the cluster.
```

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    ```bash
+    docker run --entrypoint "hdbsql" hdbsql -v
+    ```
+    Example output:
+    ```
+    HDBSQL version 2.20.20.1712178305, the SAP HANA Database interactive terminal.
+    Copyright 2000-2024 by SAP SE.
+    ```
+    3. Publish the image to a container registry.
+    4. Run the image in the Kubernetes cluster:
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:50]:
```suggestion
4. Run the image in the Kubernetes cluster:
```

File: docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    2. Test your image by running the following command:
+    ```bash
+    docker run --entrypoint "hdbsql" hdbsql -v
+    ```
+    Example output:
+    ```
+    HDBSQL version 2.20.20.1712178305, the SAP HANA Database interactive terminal.
+    Copyright 2000-2024 by SAP SE.
+    ```
+    3. Publish the image to a container registry.
```
Comment (nataliasitko) [inline docs/user/troubleshooting/03-70-cannot-connect-to-hana-db.md:49]:
```suggestion
3. Publish the image to a container registry.
```

**Review by triffer — APPROVED** [review]

**Review by triffer — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**